OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -950,6 +950,12 @@ extension OmnipodPumpManager {
             completion(.failure(PodCommsError.noPodPaired))
             return
         }
+        guard state.podState?.isFaulted == true || state.podState?.unfinalizedBolus?.scheduledCertainty == .uncertain || state.podState?.unfinalizedBolus?.isFinished != false else
```

Comment:
I don't think we need to prevent status when a bolus is running, or when there is an uncertain bolus. Does the status command not work when a pod is faulted?

Reply:
OmnipodPumpManager::getPodStatus() has never allowed a getStatus() during a certain in-progress bolus (which matches the PDM behavior which only allows for a cancel command during a bolus). Now that readPodStatus() is implemented using getDetailedStatus() (which has a 2 packet response instead of a single packet response), it would be prudent to have the same enforcement in readPodStatus() that we've always had in getPodStatus() to avoid pod comms during certain in-progress boluses when possible. I recently had a $34 fault (reset from unknown cause) on the 3rd day of use for my own personal use (which I haven't had for some time) after doing a number of the detailed "Read Pod Status" commands (both w/ and w/o active boluses) which was part of my inspiration for adding this check.

getDetailedStatus() will succeed on a faulted pod while getStatus() (what used was formally used by readPodStatus()) will not succeed. The isFaulted test in this guard simply allows Loop to fetch the DetailedStatus for a pod that has faulted mid bolus (i.e., a bolus is never actually in-progress for a faulted pod even though unfinalizedBolus might indicate that it is).

Reply:
I'm just asking you to remove the code here that prevents the pod status page from fetching fresh information while bolusing.

Reply:
Unless I am misreading something, this check is only for the heavier-weight readPodStatus() function which has to be manually invoked by the user using "Read Pod Status" and doesn't change a thing regarding pod status page refreshes. The refreshing of the pod status information uses the lighter-weight (but similar sounding) getPodStatus() function which didn't get changed in this PR which has always checked for a bolus in progress.

Reply:
Reluctantly removed the bolus checking in readPodStatus() so that hopefully this PR can land